Tests/Grid/Mapping/Metadata/MetadataTest.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace APY\DataGridBundle\Grid\Tests\Mapping\Metadata;
+
+use APY\DataGridBundle\Grid\Columns;
+use APY\DataGridBundle\Grid\Column\Column;
+use APY\DataGridBundle\Grid\Mapping\Metadata\Metadata;
+use PHPUnit\Framework\TestCase;
+
+class MetadataTest extends TestCase
+{
+    public function setUp()
+    {
+        $this->metadata = new Metadata();
+    }
+
+    public function testSetFields()
+    {
+        $field = ['foo' => 'bar'];
+
+        $this->metadata->setFields($field);
+
+        $this->assertAttributeEquals($field, 'fields', $this->metadata);
+    }
+
+    public function testGetFields()
+    {
+        $field = ['foo' => 'bar'];
+
+        $this->metadata->setFields($field);
+
+        $this->assertEquals($field, $this->metadata->getFields());
+    }
+
+    public function testSetterFieldMappingsWithField()
+    {
+        $fieldRandom = 'foo';
+        $valueRandom = 'bar';
+        $fieldMapping = [$fieldRandom => $valueRandom];
+
+        $this->metadata->setFieldsMappings($fieldMapping);
+
+        $this->assertAttributeEquals(
+            $fieldMapping,
+            'fieldsMappings',
+            $this->metadata
+        );
+    }
+
+    public function testHasFieldMappingWithField()
+    {
+        $fieldRandom = 'foo';
+        $valueRandom = 'bar';
+        $fieldMapping = [$fieldRandom => $valueRandom];
+
+        $this->metadata->setFieldsMappings($fieldMapping);
+
+        $this->assertTrue($this->metadata->hasFieldMapping($fieldRandom));
+        $this->assertFalse($this->metadata->hasFieldMapping('notAddedField'));
+    }
+
+    public function testGetterFieldMappingWithField()
```

Comment:
This can be collapsed into `testGetterMappingFieldWithType` as it is "the complete" one (you don't have execution paths in the getter that are worth of a different test)